src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/common/ParseRequestService.java

```diff
@@ -112,12 +112,13 @@ private void setAssignerId(Request request) {
             .currentRequestAttributes())
             .getRequest();
         String assignerId = httpServletRequest.getHeader("assignerId");
-        ValidationUtil.validateInputParams(UUID_PATTERN, assignerId);
 
         if (StringUtils.isBlank(assignerId)) {
             request.setAssignerId(request.getAuthenticatedUserId());
         } else {
             request.setAssignerId(UUID.fromString(assignerId));
         }
+        ValidationUtil.validateInputParams(UUID_PATTERN, request.getAssignerId().toString());
+
```

Comment:
This should be only in the else block. If it is set to AuthenticatedUserId then we don't need to do any validation.

Reply:
Done.